packages/shared/createRef/index.ts

```diff
@@ -0,0 +1,22 @@
+import type { Ref, ShallowRef } from 'vue'
+import { ref as deepRef, shallowRef } from 'vue'
+
+export type MaybeDeepRef<T = any, D extends boolean = false> = D extends true ? Ref<T> : ShallowRef<T>
```

Comment:
```suggestion
export type ShallowOrDeepRef<T = any, D extends boolean = false> = D extends true ? Ref<T> : ShallowRef<T>
```

Look for better names. But `Maybe` prefix mostly indicates "both are fine", while here it's either, so I suggest we better avoid using `Maybe`

Reply:
My one semester of Haskell is trying to get: `EitherShallowOrDeep` (`EitherShallowOrDeepRef`). But I think this is too long.